### PR TITLE
Add Spring-specific styling to Gradle Plugin's documentation

### PR DESCRIPTION
**Pull request by hsellik**
This commit replaces the default Asciidoctor styling with
Spring specific styling.

First, we need to unzip the contents of the Spring Asciidoctor
documentation resources provided by the
`io.spring.docsresources:spring-docs-resources` distribution zip. This
is done in a `/target/refdocs` folder. We then copy all files from
`src/main/asciidoc` to the same location, and then launch the generation
process.

Closes gh-15990

### CODE REVIEW

File: spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -1,34 +1,6 @@
 = Spring Boot Gradle Plugin Reference Guide
 Andy Wilkinson
-:doctype: book
```
Comment (snicoll) [inline spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/asciidoc/index.adoc]: Is this change necessary? I'd like that we keep the change to what's strictly required

Comment (DontSeeSharp) [reply spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/asciidoc/index.adoc]: Okay, I followed the example of spring-boot-docs.

Reverted by 34a545e0a878d07c7a8a03e5dc02412e385211b9.

**Comment by DontSeeSharp** [comment]
@snicoll, Thank you for the hint. 

Updated the PR updated (34a545e0a878d07c7a8a03e5dc02412e385211b9) and zip contains the docs as in actuator plugin.

**Comment by pivotal-issuemaster** [comment]
@hsellik Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=16327)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=16327) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by snicoll** [comment]
@hsellik it looks like you haven't signed the CLA. Please sign it ASAP or we'll have to revert this contribution.

**Comment by wilkinsona** [comment]
@snicoll I think that was the CLA bot making a mistake. According to https://github.com/spring-projects/spring-boot/pull/16182, @hsellik has already signed.

**Comment by snicoll** [comment]
Thanks Andy. That PR was flagged as first time contributor and I did ask rob to check our records first so something must be different (different email address?)

**Comment by wilkinsona** [comment]
They both use the same email address as far as I can tell. Perhaps the bot has been confused by all the other commits on #16182 and it's generated a false positive there for it having been signed.

**Comment by pivotal-issuemaster** [comment]
@hsellik Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=16327)!

**Comment by hsellik** [comment]
> They both use the same email address as far as I can tell. Perhaps the bot has been confused by all the other commits on #16182 and it's generated a false positive there for it having been signed.

Oh, sorry. The reason was that I changed my GitHub username. Didn't think it will have this effect.

**Comment by snicoll** [comment]
> Oh, sorry. The reason was that I changed my GitHub username. Didn't think it will have this effect.

Yeah that's probably it. Thanks for signing the CLA in any case.

**Comment by snicoll** [comment]
> Should I make a new PR so #16182 could be closed?

No need, I just synced the PR and you're all set. Thanks for your contributions so far!
